Add option to export phi-3-mini model via buck2

helunwencser opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #4190
* __->__ #4189
* #4188

as title

Differential Revision: [D59500806](https://our.internmc.facebook.com/intern/diff/D59500806/)

helunwencser committed:
Add option to export phi-3-mini model via buck2

as title

Differential Revision: [D59500806](https://our.internmc.facebook.com/intern/diff/D59500806/)

[ghstack-poisoned]

larryliu0820 commented on examples/models/phi-3-mini/export_phi-3-mini.py:
```diff
@@ -48,8 +47,14 @@ def main() -> None:
-            model, example_inputs, dynamic_shapes=dynamic_shape, strict=False
+        # TODO(lunwenh): update it to use export once
+        # https://github.com/pytorch/pytorch/issues/128394 is resolved.
+        model = torch.export._trace._export(
```
Why this though? can we still do torch.export.export()?

helunwencser replied:
`export` currently fails due to https://github.com/pytorch/pytorch/issues/128394. We will use this trick until that issue is fixed.

helunwencser committed:
Update on "Add option to export phi-3-mini model via buck2"


as title

Differential Revision: [D59500806](https://our.internmc.facebook.com/intern/diff/D59500806/)

[ghstack-poisoned]

helunwencser committed:
Update on "Add option to export phi-3-mini model via buck2"


as title

Differential Revision: [D59500806](https://our.internmc.facebook.com/intern/diff/D59500806/)

[ghstack-poisoned]

helunwencser committed:
Update on "Add option to export phi-3-mini model via buck2"


as title

Differential Revision: [D59500806](https://our.internmc.facebook.com/intern/diff/D59500806/)

[ghstack-poisoned]
